src/GovUk.Education.ExploreEducationStatistics.Content.Services/Strategies/AnalyticsWritePublicZipDownloadStrategy.cs

```diff
@@ -0,0 +1,56 @@
+#nullable enable
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Common.Utils;
+using GovUk.Education.ExploreEducationStatistics.Content.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Content.Services.Requests;
+using GovUk.Education.ExploreEducationStatistics.Content.Services.Strategies.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Content.Services.Utils;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace GovUk.Education.ExploreEducationStatistics.Content.Services.Strategies;
+
+public class AnalyticsWritePublicZipDownloadStrategy(
+    IAnalyticsPathResolver analyticsPathResolver,
+    ILogger<AnalyticsWritePublicZipDownloadStrategy> logger
+    ) : IAnalyticsWriteStrategy
+{
+    public bool CanHandle(BaseCaptureRequest request)
+    {
+        return request is CaptureZipDownloadRequest;
+    }
+
+    public async Task Report(BaseCaptureRequest request, CancellationToken cancellationToken)
+    {
+        var zipDownloadRequest = request as CaptureZipDownloadRequest
+                                 ?? throw new ArgumentException($"request isn't a {nameof(CaptureZipDownloadRequest)}");
+
+        logger.LogInformation(
+            "Capturing {RequestTypeName} for releaseVersion {ReleaseVersionId}{WithSubject}",
+            zipDownloadRequest.GetType().ToString(),
+            zipDownloadRequest.ReleaseVersionId,
+            zipDownloadRequest.SubjectId == null
+                ? ""
+                : " for subject " + zipDownloadRequest.SubjectId);
+
+        var serialisedRequest = JsonSerializationUtils.Serialize(
+            obj: zipDownloadRequest,
+            formatting: Formatting.Indented,
+            orderedProperties: true,
+            camelCase: true);
+
+        var directory = analyticsPathResolver.PublicZipDownloadsDirectoryPath();
+
+        var filename =
```

Comment:
I'm assuming that adding the random string to the end of the filename successfully stopped the invalid JSON being produced in that other bug?

Reply:
Nice one - fingers crossed then, we should never *actually* end up with invalid JSON!